### ISSUE TITLE
Add popup WebView2 window method

### ISSUE THREAD

**Issue by kingcean** [MicrosoftEdge/WebView2Feedback#2221]
Currently we can use WebView2 control in WPF, UWP, Windows App SDK, etc. to load a web page embedded. It could be better if provide a method to popup a window with WebView2 and other default components like js function `window.open()` does but it is in host app side, such as following. So that we can call this method when we need popup a web window and no more customization requirement.

```diff
namespace Microsoft.Web.WebView2.Core;
public sealed class CoreWebView2
{
+  public static Window OpenWindow(string uri, CoreWebView2WindowFeatures windowFeatures = null);
+  public static Window OpenWindowToString(string htmlContent, CoreWebView2WindowFeatures windowFeatures = null);
  // Other members.
}
```

**Comment by quietfanatic** [comment]
That's an interesting idea.  I'm not sure I see a use case that wouldn't be satisfied by simply opening a link in the OS's default browser.  Can you share more about your scenario?

**Comment by kingcean** [comment]
Currently, what we can do to open a link in a new window is either to open OS's default browser or to open a customized window with WebView2 element. Each of them owns its disadvantage in some cases.

- "Open OS's default browser" means we can't share cookies between OS's default browser and WebView2;
- "Open a customized window with WebView2 element" means we need do more for the new window embedded webpage.

So, add a method to open a WebView2 window with the source URI and `CoreWebView2WindowFeatures` arguments can let us share cookies without additional development.

**Comment by champnic** [comment]
Thanks for the feature request @kingcean! You can do this by running `window.open` using `ExecuteScriptAsync` pretty simply, so I don't think we'd add an extra API at this point. Or do you mean even before initializing a WebView2, this would create a new WebView2 and put it into a new window?

**Comment by kingcean** [comment]
> Or do you mean even before initializing a WebView2, this would create a new WebView2 and put it into a new window?

Yes. For example, we have a window with a native button to open a new window which only includes a WebView2 element and other accessories defined by `CoreWebView2WindowFeatures` options. The webpage loaded by this WebView2 shares cookies and other local data from/to other pages loaded by WebView2 in history/future. So, we cannot call `window.open` function in JS by because there is no WebView2 instance existed.

**Comment by quietfanatic** [comment]
A moderately awkward workaround would be to create an invisible webview just to call the script on it.

**Comment by champnic** [comment]
@kingcean If you use the same user data folder when creating a new WebView2, then cookies and other data will be shared automatically. It should generally be pretty easy to open a window to the specification you need it with just a WebView2 in it. Do you have scenarios where that's prohibitive?

**Comment by kingcean** [comment]
> If you use the same user data folder when creating a new WebView2, then cookies and other data will be shared automatically.

Yes.

> Do you have scenarios where that's prohibitive?

No.

---

So the feature request here is to add a static/helper method to open a new window with a WebView2 element (and related accessories) as same as what JS script `window.open` does but does not need an existing WebView2 already there anymore at that moment.

Currently, the workaround is to create an invisible WebView2 to call the JS script (high memory cost), or to add a window with element and logic of WebView2 (more dev effort).

**Comment by champnic** [comment]
Thanks - I've added this scenario to our backlog :)

**Comment by ajtruckle** [comment]
Is this related to #2376 ? Because it sounds ideal. A simple way for me to display view-source URL in a popup window.

**Comment by champnic** [comment]
This isn't specifically related to #2376, but could be used in that manner. You can also use the devtools to display source, or get the source and open it in a new window manually.
